src/PlayerControls.cpp

```diff
@@ -35,6 +33,7 @@ void PlayerControlsSystem::on_tick()
         }
         if (keys.is_down(bindings.right))
         {
+            player.fsm.change_state(PlayerTransition::WALK);
             replace_component<Flip>(player_entity, None);
```

Comment:
It just occurred to me: All of this code about what to do with the player could be inside the `on_enter` and `on_exit` functions, otherwise, what are those doing?